doc/architecture-decisions/0018-route-crd-and-contour-controller.md

```diff
@@ -0,0 +1,48 @@
+# 18. Add Contour as an Alternative Ingress Solution Provider
+
+Date: 2020-10-09
+
+## Status
+
+Proposal
+
+## Context
+In our efforts to allow more optionality in ingress solutions for CF-for-K8s, we
+want to allow the use of Contour as a potential alternative to Istio.
+
+### Proposed Design
+
+We propose:
+1. Adding a configuration option `ingress_solution_provider` to CF-for-K8s, with
+   potential values `istio` or `contour`.
+2. Creating another kubebuilder controller, similar to routecontroller, to
+   handle when `ingress_solution_provider: contour` is configured.
+3. Renaming `Route Controller` to `Istio Controller`
+
+![Proposed ContourControllerDesign](../assets/contour-controller.png)
+
+### Open Questions
+1. What do we do about the config? Does contour config live in cf-k8s-networking
+   and eventually it moves to cf-for-k8s?
+2. What happens if an operator wants to change their ingress solution provider,
+   will they have to redeploy CF-for-K8s? Is that a big deal?
```

Comment:
If we have a single controller watching for configmap changes on the mounted volume it can detect configmap change and reload itself. Or also, I saw one Prometheus deployment with a sidecar which restarts Prometheus when configmap changes.